source/extensions/filters/listener/tls_inspector/tls_inspector.cc

```diff
@@ -184,6 +204,9 @@ ParseState Filter::parseClientHello(const void* data, size_t len) {
     } else {
       config_->stats().tls_not_found_.inc();
     }
+    // Record client_hello size as we're done processing.
+    config_->stats().client_hello_size_.recordValue(
+        computeClientHelloSize(bio.get(), bytes_already_processed, len));
     return ParseState::Done;
   default:
     return ParseState::Error;
```

Comment:
should we track here?  how about having this and the above break out of the switch and consistently track bytes before returning ParseState::Error

Reply:
I do think consistently tracking a catch all is slightly better coding hygiene though.  Any reason not to?

Reply:
I mean FWIW I agree looking at the ssl_read documentation there's _probably_ not other errors (WANT_WRITE is possible but unlikely?) but just as a code style thing we can have correctness rely on that being accurate and not changing, or we can track the stat outside the switch =P

Reply:
Done.